src/arch/cuda/hybridAPI/hapi_impl.cpp

```diff
@@ -231,8 +312,97 @@ void hapiMapping(char** argv) {
       CmiAbort("Unsupported mapping type!");
   }
 
-  // Set device for each PE
+  // Set device and store PE-device mapping
   hapiCheck(cudaSetDevice(CpvAccess(my_device)));
+#if CMK_SMP
+  CmiLock(CsvAccess(gpu_manager).device_mapping_lock);
+#endif
+  CsvAccess(gpu_manager).device_map.emplace(CmiMyPe(),
+      &(CsvAccess(gpu_manager).device_managers[CpvAccess(my_device)]));
+#if CMK_SMP
+  CmiUnlock(CsvAccess(gpu_manager).device_mapping_lock);
+#endif
+
+  // Process device communication buffer parameters (in MB)
+  int input_comm_buffer_size;
+  if (CmiGetArgIntDesc(argv, "+gpucommbuffer", &input_comm_buffer_size,
+        "GPU communication buffer size (in MB)")) {
+    if (CmiMyRank() == 0) {
+      // Round up size to the closest power of 2
+      size_t comm_buffer_size = (size_t)input_comm_buffer_size * 1024 * 1024;
+      int size_log2 = std::ceil(std::log2((double)comm_buffer_size));
+      CsvAccess(gpu_manager).comm_buffer_size = (size_t)std::pow(2, size_log2);
```

Comment:
What happens to this value when the user doesn't pass `+gpucommbuffer`? Is there a reasonable default chosen?

Reply:
It's 64MB by default (see `gpumanager.h`).